server/src/routes/api/v1/userSaveFileRouter.js

```diff
@@ -17,23 +22,75 @@ userSaveFileRouter.get("/:id", async (req, res) => {
 
 userSaveFileRouter.post("/:id", async (req, res) => {
   const body = req.body
-  const puzzlesId = req.params.id
+  const puzzleId = req.params.id
   const userId = req.user.id
+  let correctlySolved = false
   
-  const formInput = cleanUserInput(body)
+  const formInput = cleanUserSubmission(body)
   let serializedformInput = JSON.stringify(formInput);
 
   let userSave = {}
   userSave.savedPuzzle = serializedformInput
   userSave.userId = userId
-  userSave.puzzlesId = puzzlesId
+  userSave.puzzleId = puzzleId
 
-  try {
-    const newSave = await UserSaveFile.query().insertAndFetch(userSave)
-    return res.status(200).json({ userSaveFile: newSave })
-  } catch(error) {
-    return res.status(500).json({ error })
+  const user = await User.query().findById(userId)
+  const userSaveFile = await user.$relatedQuery("userSaveFile")
+
+  let matchedSave = false
+  userSaveFile.forEach(save => {
+    if(save.puzzleId === puzzleId){
+      matchedSave = save.id
+    }
+  })
```

Comment:
this can be refactored. Rather than querying for all related userSaveFiles, we can do something along the lines of"

```
const UserSaveFile.query().findOne(userId: userId, puzzleId: puzzleId) 
```